### PR TITLE
Remove lookupSymbol() and have all callers use SymbolInfo::lookup() instead

### PR DESCRIPTION
**Pull request by grynspan**
This PR removes the free function `lookupSymbol()` and modifies its callers to use `SymbolInfo::lookup()` instead.

In a previous PR, I refactored `lookupSymbol()` to call through to `SymbolInfo::lookup()` which itself returned an `llvm::Optional<SymbolInfo>`, which `lookupSymbol()` unpacks and copies into an out-param. As discussed in the review for that PR, we might as well remove the stub function and update callers to just use the underlying function.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

File: stdlib/public/runtime/Errors.cpp [hunk 1]
```diff
@@ -144,18 +144,16 @@ static bool getSymbolNameAddr(llvm::StringRef libraryName,
 void swift::dumpStackTraceEntry(unsigned index, void *framePC,
                                 bool shortOutput) {
 #if SWIFT_STDLIB_SUPPORTS_BACKTRACE_REPORTING && SWIFT_STDLIB_HAS_DLADDR
-  SymbolInfo syminfo;
-
-  // 0 is failure for lookupSymbol
-  if (0 == lookupSymbol(framePC, &syminfo)) {
+  auto syminfo = SymbolInfo::lookup(framePC);
```
Comment (compnerd) [inline stdlib/public/runtime/Errors.cpp:147]: I'm not sure `auto` is great here - you don't have any indication that `SymbolInfo::lookup` returns a `SymbolInfo`.  Similar throughout.

**Review by compnerd — APPROVED** [review]

File: stdlib/public/runtime/Errors.cpp (hunk 1 above)
Comment (grynspan) [reply stdlib/public/runtime/Errors.cpp:147]: It doesn't, it returns an `llvm::Optional<SymbolInfo>`. This use of `auto` is not unheard of in the repo and is akin to Swift's `let` (well, `var`) which also doesn't explicitly state the type of the value.

Do you feel strongly about changing it?

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

**Review by al45tair — APPROVED** [review]
LGTM. I do have reservations about `auto` (I don't like that it hides the type at the call site, which makes the code much harder to read because now you have to find the declaration to understand what e.g. `getFilename()` might refer to), _but_ I think I'd also accept that using it is idiomatic C++ these days and I'm doubtless guilty of using it myself in places where later on I might wish I hadn't.

**Comment by grynspan** [comment]
> LGTM. I do have reservations about `auto` (I don't like that it hides the type at the call site, which makes the code much harder to read because now you have to find the declaration to understand what e.g. `getFilename()` might refer to), _but_ I think I'd also accept that using it is idiomatic C++ these days and I'm doubtless guilty of using it myself in places where later on I might wish I hadn't.

I've got another PR coming up to enable `getFilename()` on Windows, and it requires some refactoring anyway to avoid use-after-free scenarios, so I'll revisit my use of `auto` in that one.

**Pull request merged**